vendors/espressif/boards/esp32/aws_tests/application_code/esp_smp_tests.c

```diff
@@ -0,0 +1,141 @@
+/*
+ * Amazon FreeRTOS V1.1.4
+ * Copyright (C) 2018 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * http://aws.amazon.com/freertos
+ * http://www.FreeRTOS.org
+ */
+
+
+/* FreeRTOS includes. */
+#include "FreeRTOS.h"
+#include "task.h"
+#include "semphr.h"
+
+/* Test framework includes. */
+#include "unity_fixture.h"
+
+#define ESP_TEST_DEFAULT_PRIORITY (tskIDLE_PRIORITY + 1)
+#define ESP_TEST_HIGH_PRIORITY (tskIDLE_PRIORITY + 3)
+#define ESP_TASK_DEFAULT_STACK_SIZE    (configMINIMAL_STACK_SIZE * 4)
+
+#define ESP_TEST_PINNED_TASK_NAME "ESP_Pin"
+#define TASK_CREATION_TIMEOUT pdMS_TO_TICKS(1000)
+/**
+ * @brief Test group for ESP Task creation tests
+ */
+TEST_GROUP( ESP_SMP_Task_Creation );
+
+/**
+ * @brief Test group runner for ESP SMP task creation
+ */
+TEST_GROUP_RUNNER( ESP_SMP_Task_Creation )
+{
+    RUN_TEST_CASE( ESP_SMP_Task_Creation, IsMultiCore );
+    RUN_TEST_CASE( ESP_SMP_Task_Creation, CreatePinnedTaskOnCore0 );
+    RUN_TEST_CASE( ESP_SMP_Task_Creation, CreatePinnedTaskOnCore1 );
+}
+
+TEST_SETUP( ESP_SMP_Task_Creation) {}
+TEST_TEAR_DOWN( ESP_SMP_Task_Creation) {}
+
+/**
+ * @brief The response structure from report core
+ */
+typedef struct {
+    SemaphoreHandle_t done; /*< The reportCore task sets this semaphore on completion */
+	BaseType_t			xCoreID; /*< The CPU ID on which report Core ran */
+} coreReport_t;
+
+/**
+ * @brief Record the core on which this task runs, then suspend
+ */
+void reportCore( void* pvReport ) {
```

Comment:
I think it's better to make this function static to limit scope. Any thoughts?

Reply:
Yes. Thanks for noting. Fixed.